src/lib.rs

```diff
@@ -754,7 +754,11 @@ impl<'a, L: 'static + Ledger> KeystoreState<'a, L> {
                 summary
                     .received_memos
                     .extend(outputs.iter().map(|(memo, _, uid, _)| (memo.clone(), *uid)));
-                for key in &model.sending_accounts.keys() {
+                for key in &model
+                    .sending_accounts
+                    .iter_keys()
+                    .collect::<Vec<UserKeyPair>>()
```

Comment:
Ah, do we still need a `collect` here to ensure `model.sending_accounts` is not borrowed when we call `try_open_memos`? That's a bit unfortunate. But probably unavoidable. Oh well.

Reply:
Yeah, the `collect` is still needed. 😥